contracts/staking/src/contract.rs

```diff
@@ -524,7 +523,11 @@ pub fn deposit(mut deps: DepsMut, info: MessageInfo, env: Env) -> Result<Respons
 
     validate(&config, &info, &env, vec![Verify::NonZeroSingleInfoFund])?;
 
-    check_slashing(&mut deps, &env)?;
+    let all_delegations = deps
```

Comment:
Can this not go inside the check_slashing function?

Reply:
i want to avoid calling this muliple times. I use this in `get_validator_for_deposit`. I pass it as a reference to that func so as to avoid calling the deps.querier.query_all_delegations thing. The reason why our gas costs are so high is because we were querying the blockchain too many times.